### PR TITLE
Kk 107 events

### PR DESCRIPTION
**Pull request by DiegoRamms**

### CODE REVIEW

File: core/data/src/main/java/com/kk/data/SocketServiceImp.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                url(
+                    when (userType) {
+                        UserType.HostType -> BaseUrl.urlHost
+                        UserType.PlayerType -> BaseUrl.urlPlayer
+                    }
+                )
+            }
+
+            if (socket?.isActive == true) {
+                BaseResult.Success("Connected")
```
Comment (dmroldanch) [inline core/data/src/main/java/com/kk/data/SocketServiceImp.kt]: You need to create a file with your messages, you can use const

File: app/src/main/java/com/kk/knowledgeknockout/MainActivity.kt [hunk 2]
```diff
@@ -1,32 +1,46 @@
...
+    var data: StateFlow<BaseResponse<String>>? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         setContent {
             KnowledgeKnockoutTheme {
+                val state = data?.collectAsState()
+
+                Log.e("state", state.toString())
```
Comment (dmroldanch) [inline app/src/main/java/com/kk/knowledgeknockout/MainActivity.kt]: Remove Logs

File: core/data/src/main/java/com/kk/data/SocketServiceImp.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        } catch (e: Exception) {
+            flow {
+                emit(BaseResult.Error(e))
+            }
+        }
+    }
+
+    override suspend fun closeSocket() {
+        socket?.close(CloseReason(code = CloseReason.Codes.NORMAL, "Closed"))
```
Comment (dmroldanch) [inline core/data/src/main/java/com/kk/data/SocketServiceImp.kt]: You need to create a file with your messages, you can use const

File: presentation/src/main/java/com/kk/presentation/host/creategame/CreateRoomViewModel.kt [hunk 4]
```diff
@@ -1,22 +1,78 @@
...
+class CreateRoomViewModel(private val socketServiceHost: ISocketService) : ViewModel()
+//BaseViewModel<CreateRoomContract.Event, CreateRoomContract.State, CreateRoomContract.Effect>()
+{
+
+    private val _uiState = MutableStateFlow(BaseResponse("",""))
+    val uiState get() = _uiState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            //socketServiceHost.connectSocket()
```
Comment (dmroldanch) [inline presentation/src/main/java/com/kk/presentation/host/creategame/CreateRoomViewModel.kt]: You don't push commented code

File: settings.gradle [hunk 5]
```diff
@@ -26,8 +26,14 @@ dependencyResolutionManagement {
                 version('koin-version','3.3.2')
                 version('lottieVersion', '5.2.0')
                 version('ktorVersion', '1.5.0')
+                version('gsonVersion', '2.10.1')
+                version('androidCompose','3.4.1')
+
+                library('koinAndroid','io.insert-koin','koin-android').versionRef('koin-version')
+                //library('koin-compose','io.insert-koin','koin-compose').versionRef('koin-version')
```
Comment (dmroldanch) [inline settings.gradle]: Remove those libraries  if you don't use them

File: core/data/src/main/java/com/kk/data/SocketServiceImp.kt (hunk 1 above)
Comment (DiegoRamms) [reply core/data/src/main/java/com/kk/data/SocketServiceImp.kt]: resolved 👍

File: app/src/main/java/com/kk/knowledgeknockout/MainActivity.kt (hunk 2 above)
Comment (DiegoRamms) [reply app/src/main/java/com/kk/knowledgeknockout/MainActivity.kt]: resolved 👍

File: core/data/src/main/java/com/kk/data/SocketServiceImp.kt (hunk 3 above)
Comment (DiegoRamms) [reply core/data/src/main/java/com/kk/data/SocketServiceImp.kt]: resolved 👍

File: presentation/src/main/java/com/kk/presentation/host/creategame/CreateRoomViewModel.kt (hunk 4 above)
Comment (DiegoRamms) [reply presentation/src/main/java/com/kk/presentation/host/creategame/CreateRoomViewModel.kt]: resolved 👍

File: settings.gradle (hunk 5 above)
Comment (DiegoRamms) [reply settings.gradle]: resolved 👍

**Review by dmroldanch — APPROVED** [review]
Good Job 🥇 👍

**Pull request merged**